### PR TITLE
Update docs on the measuring performance page

### PR DESCRIPTION
**Pull request by jakemac53**
Closes https://github.com/dart-lang/build/issues/3648

@FufferKS does this help?

### CODE REVIEW

File: docs/measuring_performance.md [hunk 1]
```diff
@@ -33,3 +33,56 @@ The time for each action is split into at 3 primary pieces:
...
+**Stage Time**: The start and end time (relative to the start of the build) for
+the stage of the time slice you are currently hovering over.
+
+**Slice Time**: The total time spent on the current "slice", if
+[show async slices](#show-async-slices) is enabled. These are actual slices of
+synchronous work, up to the granularity of the
+[slices resolution](#slices-resolution) configuration.
+
+**User time**: The actual time spent synchronously performing actions related to
+the current stage. This does not count time spent waiting for asynchronous taks.
```
Comment (FufferKS) [inline docs/measuring_performance.md:51]: typo taks -> tasks

File: docs/measuring_performance.md [hunk 2]
```diff
@@ -33,3 +33,67 @@ The time for each action is split into at 3 primary pieces:
...
+**Stage time**: The start and end time (relative to the start of the build) for
+the stage of the time slice you are currently hovering over.
+
+**Slice time**: The start and end time (relative to the start of the build) for
+the slice.
+
+**Slice duration**: The total time spent on the current slice.
+
+**User time**: The actual time spent synchronously performing actions related to
+the current stage. This does not count time spent waiting for asynchronous taks.
```
Comment (jakemac53) [inline docs/measuring_performance.md:62]:
```suggestion
the current stage. This does not count time spent waiting for asynchronous tasks.
```

File: docs/measuring_performance.md [hunk 3]
```diff
@@ -33,3 +33,67 @@ The time for each action is split into at 3 primary pieces:
...
+(unless it is less than the slice resolution).
+
+If [show async slices](#show-async-slices) is disabled, the granularity is
+infinite, so a slice is just the entire time spent in a stage, including all
+async work.
+
+### Stage time, slice time, user time, and real time
+
+Within each action (row), you can hover over one of the active time slices,
+which will show you three time metrics, described as follows:
```
Comment (jakemac53) [inline docs/measuring_performance.md:51]:
```suggestion
which will show you three several metrics, described as follows:
```

**Comment by FufferKS** [comment]
Amazing, thanks for replying on such a short notice!

This is definitely helpful, if you don't mind I will ask some more in depth questions tomorrow.

File: docs/measuring_performance.md [hunk 4]
```diff
@@ -33,3 +33,67 @@ The time for each action is split into at 3 primary pieces:
...
+(unless it is less than the slice resolution).
+
+If [show async slices](#show-async-slices) is disabled, the granularity is
+infinite, so a slice is just the entire time spent in a stage, including all
+async work.
+
+### Stage time, slice time, user time, and real time
+
+Within each action (row), you can hover over one of the active time slices,
+which will show you three several metrics, described as follows:
```
Comment (natebosch) [inline docs/measuring_performance.md:51]:
```suggestion
which will show several metrics, described as follows:
```

File: docs/measuring_performance.md [hunk 5]
```diff
@@ -33,3 +33,67 @@ The time for each action is split into at 3 primary pieces:
...
+
+**Real time**: The total time between the start and end time of this stage. This
+includes time waiting for asynchronous tasks.
+
+## Configuring the timeline
+
+### Hide skipped actions
+
+Many actions in a build are "optional", which means they only run if some other
+action asks to read their output. This setting toggles whether those rows
```
Comment (natebosch) [inline docs/measuring_performance.md:74]: Double checking - is skipped only the optional assets, or does it include not-invalidated assets?

**Review by natebosch — APPROVED** [review]

**Pull request merged**